### PR TITLE
Lastline csv bug

### PR DESCRIPTION
**Pull request by yucohen**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44728

## Description
csv files requires different http request.
also deleted the note from the readme that said csv is not supported.
while on the call with the customer, discovered that eml files also fail the command. the csv handling did not help in this case, so also added that in note to the readme.

### CODE REVIEW

**Review by yaakovi — APPROVED** [review]

**Comment by yucohen** [comment]
@jrosenpanw hi
Can someone please have a look?
thanks

**Comment by yucohen** [comment]
@ShirleyDenkberg Hi 
can someone please have a look?
thanks

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

File: Packs/Lastline/ReleaseNotes/1_0_10.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
 #### Integrations
 ##### Lastline v2
-- Fixed an issue where **lastline-upload-file** failed when given a csv file.
+- Fixed an issue where the **lastline-upload-file** command failed when given a CSV file.
```
Comment (yaakovi) [inline Packs/Lastline/ReleaseNotes/1_0_10.md:4]:
```suggestion
- Fixed an issue where the ***lastline-upload-file*** command failed when given a CSV file.
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 2]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
     client = Client(base_url=params.get('url'), api_params=api_params, credentials=params.get('credentials'))
     with pytest.raises(DemistoException, match='Authentication Error.'):
         client.test_module_command()
+
+
+@pytest.mark.parametrize('params', [{'url': 'testurl.com',
+                                     'credentials': {'identifier': 'identifier',
+                                                     'password': 'password'}}])
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:154]: No need to use `parametrize` here.

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 3]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+def test_csvs_handles_correctly(mocker, params):
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:155]: Improve function name.
The name should include the name of the unit it tests (command/function) and the use case.

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 4]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+    """
+        Given:
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:157]:
```suggestion
    Given:
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 5]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+        - Valid params and a csv file to upload
+
+    When:
+        - Running the integration upload-file command
+
+    Then:
+        - Validating that handle_csv is being called
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:164]: Check that there was a call to the login API.

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 6]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+    """
+    client = Client(base_url=params.get('url'), credentials=params.get('credentials'), api_params={'EntryID': '1234@'})
+    mocker.patch.object(demisto, 'getFilePath', return_value={'id': id, 'path': 'test/test.txt', 'name': 'test.csv'})
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:167]:
```suggestion
    mocker.patch.object(demisto, 'getFilePath', return_value={'id': id, 'path': 'test/test.csv', 'name': 'test.csv'})
```

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 7]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+    mocker.patch.object(os.path, 'splitext', return_value=['test', '.csv'])
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:168]: Redundant

File: Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py [hunk 8]
```diff
@@ -146,3 +147,27 @@ def test_test_module_wrong_credentials(mocker, params):
...
+    mocker.patch.object(Lastline_v2, 'file_hash', return_value='dummyhash1234')
+    # if test failed, then this line wasn't effective, meaning handle_csv was not being called
+    mocker.patch.object(client, 'handle_csv', return_value={})
```
Comment (yaakovi) [inline Packs/Lastline/Integrations/Lastline_v2/Lastline_v2_test.py:171]: Mock the requests instead and check the login endpoint was called.

**Review by yaakovi — CHANGES_REQUESTED** [review]

**Pull request merged**